R/utils.R

```diff
@@ -81,12 +81,30 @@ convertWildCards <- function(val) {
   gsub("_", "(.)", val)
 }
 
-getSelector <- function(key, value) {
+getSelector <- function(key, value, exact) {
+  if ( is.null(value ) ) {
+    return("")
+  }
   comp <- " = "
-  if (isPatternValue(value)) {
+  if (( value == "%" || !exact ) &&
+      isPatternValue(value)) {
     comp <- " LIKE "
   }
+  if (exact && (value != "%"))
+    value <- escapeChars(value)
   value <- paste0("'", value, "'")
 
   paste0(" AND ", key, comp, value)
 }
+
+# Will iterate over charsToEsc argument and for each:
+# will escape any un-escaped occurance in `x`.
+escapeChars <- function(x, charsToEsc = c("_"), escChar ="\\\\") {
+  if(is.null(x) || inherits(x, "AsIs")) return(x)
+  for (ch in charsToEsc) {
+    pattern <- paste0("([^", escChar, "])(", ch,")")
```

Comment:
If there's an escape character in the string, doesn't that imply that the string is already escaped? 

Also I think you could drop the loop in favour of `gsub("[^_%]", "\\\\\\1", x)`?

Reply:
Thanks Hadley.

* I was trying to be robust here and cover the case when someone does something silly like pass
`THIS\\_IS_MY\\_HALF_ESCAPED\\_TABLE`.
* Fixed the loop, along the lines of your suggestion - thanks!